src/cli.nim

```diff
@@ -0,0 +1,149 @@
+import std/[options, os, parseopt, strformat, strutils, terminal]
+
+type
+  Action* = enum
+    actSync, actCheck
+
+  Mode* = enum
+    modeChoose, modeIncludeMissing, modeExcludeMissing
+
+  Verbosity* = enum
+    verQuiet, verNormal, verDetailed
+
+  Conf* = object
+    action*: Action
+    exercise*: Option[string]
+    mode*: Mode
+    verbosity*: Verbosity
+
+  Opt = enum
+    optExercise, optCheck, optMode, optVerbosity, optHelp, optVersion
+
+  OptKey = tuple
+    short: string
+    long: string
+
+const
+  NimblePkgVersion {.strdefine.}: string = "unknown"
+
+  optKeys: array[Opt, OptKey] = [
+    ("e", "exercise"),
+    ("c", "check"),
+    ("m", "mode"),
+    ("o", "verbosity"),
+    ("h", "help"),
+    ("v", "version"),
+  ]
+
+  optsNoVal = {optCheck, optHelp, optVersion}
+
+func short(opt: Opt): string =
```

Comment:
When should a `func` be used over ` proc`? I think I remember it being about side-effects.

Reply:
Yes - we can use `func` instead of `proc` whenever the proc doesn't have side effects. In my opinion, it increases readability if we consisently use `func` whenever possible in a codebase. I'll eventually do that in this repo as I refactor the code.

See also:
- https://nim-lang.org/docs/manual.html#pragmas-nosideeffect-pragma
- https://nim-lang.org/blog/2020/09/01/write-tracking.html